File: ios/ColorPickerViewManager.m
//
//  ColorPickerViewManager.m
//  Toddle
//
//  Created by Raj Jakasaniya on 08/07/20.
//


#import <React/RCTViewManager.h>
#import <React/RCTConvert.h>

@interface
RCT_EXTERN_MODULE(ColorPickerViewManager, RCTViewManager)
RCT_EXTERN_METHOD(show:(nonnull NSNumber *)node)
RCT_EXTERN_METHOD(hide:(nonnull NSNumber *)node)
RCT_EXPORT_VIEW_PROPERTY(onSelected, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onCancelled, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onColorChange, RCTDirectEventBlock)
@end

File: ios/ColorPickerViewManager.swift
//
//  ColorPickerManager.swift
//  Toddle
//
//  Created by Raj Jakasaniya on 08/07/20.
//

import Foundation
import UIKit

@objc(ColorPickerViewManager)
class ColorPickerViewManager: RCTViewManager, ColorPickerDelegate {
  var colorPicker: ColorPickerView!

  override func view() -> UIView! {
    colorPicker = ColorPickerView()
    colorPicker.frame = CGRect(x: 200, y: 200, width: 200, height: 200)
    colorPicker.delegate = self
    colorPicker.layer.cornerRadius = colorPicker.frame.width/2
    colorPicker.clipsToBounds = true
    colorPicker.layer.borderColor = UIColor.black.cgColor
    colorPicker.layer.borderWidth = 4
    return colorPicker
  }
  
  func ColorColorPickerTouched(sender: ColorPickerView, color: UIColor, point: CGPoint, state: UIGestureRecognizer.State) {
    colorPicker.setColor(color: color)
  }
  

  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  @objc public func show(_ node: NSNumber) {
    colorPicker.isHidden = false
  }
  @objc public func hide(_ node: NSNumber) {
    colorPicker.isHidden = true
  }
  
  
}


